### PR TITLE
Add KYC rule action

### PR DESCRIPTION
**Pull request by colinbird**

### CODE REVIEW

File: spec/definitions/Rules/Actions/kyc.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        type: string
+        description: What to do if verification is before processing, and is rejected
+        enum:
+          - process
+          - decline
+          - use-alternate-gateway
+        default: decline
+      alternate-gateway-if-rejected:
+        type: string
+        description: Gateway account to use if use-alternate-gateway is selected for alternate-gateway-if-rejected
```
Comment (colinbird) [inline spec/definitions/Rules/Actions/kyc.yaml]: I wonder if `if-rejected-before` should be an option which contains both the enum and the gateway account id

File: spec/definitions/Rules/Actions/kyc.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+description: "Verify customer identity"
+allOf:
+  - $ref: "#/definitions/RuleAction"
+  - type: object
+    properties:
+      exclude:
```
Comment (colinbird) [inline spec/definitions/Rules/Actions/kyc.yaml]: excludePolicy?

**Comment by colinbird** [comment]
devs please review

**Comment by colinbird** [comment]
devs please review

File: spec/definitions/Rules/RuleAction.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@ properties:
       - trigger-webhook
       - stop-subscriptions
       - add-risk-score
+      - kyc
```
Comment (dara123) [inline spec/definitions/Rules/RuleAction.yaml]: should this be `request-kyc`?

Comment (colinbird) [reply spec/definitions/Rules/RuleAction.yaml]: yes

File: spec/definitions/Rules/Actions/request-kyc.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        default: customers-with-accepted-document
+      isMandatory:
+        type: boolean
+        description: Is the verification mandatory
+        default: true
+      promptPolicy:
+        type: string
+        description: When to prompt, before or after processing the transaction
+        enum:
+          - before-txn-process
```
Comment (dara123) [inline spec/definitions/Rules/Actions/request-kyc.yaml]: maybe `transaction` not `txn`

**Pull request merged**